src/main/java/com/nylas/Neural.java

```diff
@@ -0,0 +1,105 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Neural {
+	protected final String neuralPath = "neural";
+	protected final NylasAccount account;
+	protected final NylasClient client;
+	protected final String accessToken;
+	private static final Pattern imagePattern = Pattern.compile("[(']cid:(.)*[)']");
+
+	public Neural(NylasAccount account, NylasClient client, String accessToken) {
+		this.account = account;
+		this.client = client;
+		this.accessToken = accessToken;
+	}
+
+	public List<NeuralSentimentAnalysis> sentimentAnalysisMessage(List<String> messageIds)
+			throws RequestFailedException, IOException {
+		Map<String, Object> body = new HashMap<>();
+		body.put("message_id", messageIds);
+		Type listType = JsonHelper.listTypeOf(NeuralSentimentAnalysis.class);
+		return neuralRequest("sentiment", body, listType);
+	}
+
+	public NeuralSentimentAnalysis sentimentAnalysisText(String text) throws RequestFailedException, IOException {
+		Map<String, Object> body = new HashMap<>();
+		body.put("text", text);
+		return neuralRequest("sentiment", body, NeuralSentimentAnalysis.class);
+	}
+
+	public List<NeuralSignatureExtraction> extractSignature(List<String> messageIds)
+			throws RequestFailedException, IOException {
+		return extractSignature(messageIds, null);
+	}
+
+	public List<NeuralSignatureExtraction> extractSignature(List<String> messageIds, NeuralMessageOptions options)
+			throws RequestFailedException, IOException {
+		Map<String, Object> body = new HashMap<>();
+		body.put("message_id", messageIds);
+		if(options != null) {
+			body.putAll(options.toMap());
+		}
+		Type listType = JsonHelper.listTypeOf(NeuralSignatureExtraction.class);
+		return neuralRequest("signature", body, listType);
+	}
+
+	public NeuralOcr ocrRequest(String fileId, int... pages) throws RequestFailedException, IOException {
+		Map<String, Object> body = new HashMap<>();
+		body.put("file_id", fileId);
+		if(pages != null) {
+			body.put("pages", pages);
+		}
+		return neuralRequest("ocr", body, NeuralOcr.class);
+	}
+
+	public List<NeuralCleanConversation> cleanConversation(List<String> messageIds)
+			throws RequestFailedException, IOException {
+		return cleanConversation(messageIds, null);
+	}
+
+	public List<NeuralCleanConversation> cleanConversation(List<String> messageIds, NeuralMessageOptions options)
+			throws RequestFailedException, IOException {
+		Map<String, Object> body = new HashMap<>();
+		body.put("message_id", messageIds);
+		if(options != null) {
+			body.putAll(options.toMap());
+
+			// Remove parse_contacts key as this option is only for signature extraction
+			if(body.get("parse_contacts") != null) {
+				body.remove("parse_contacts");
+			}
+		}
+		Type listType = JsonHelper.listTypeOf(NeuralCleanConversation.class);
+		return neuralRequest("conversation", body, listType);
+	}
+
+	public List<File> extractImages(NeuralCleanConversation neuralCleanConversation) throws RequestFailedException, IOException {
```

Comment:
While it would be great to see more javadoc on much of the new code, I think especially methods like this that don't map one-to-one to api requests would be good to document.

Reply:
I wanted to do that before requesting a PR but forgot! Will add them now.